src/controllers/likeController.js

```diff
@@ -0,0 +1,127 @@
+import * as ApplicationError from '../utils/errors/applicationsErrors';
+import { UpdateOrCreate, getLikes } from '../services/likeService';
+import { Accomodation, Like} from '../database/models';
+
+const like = async (req, res) => {
+  try {
+    const { accomodation_id: accomodationId } = req.params;
+    const { id: userId } = req.user;
+    const accomodationExist = await Accomodation.findOne({ where: { id: accomodationId } });
+    if (!accomodationExist) {
+      return ApplicationError.notFoundError(
+        'This accomodation does not exist',
+        res
+      );
+    }
+    const { created } = await UpdateOrCreate(Like, {accomodationId, userId}, true);  // UpdateOrCreate(model, {item}, flag) if flag is true means like unless it is dislike
+    if (created) {
+      return res
+        .status(201)
+        .json({
+          status: 201,
+          data:{
+          message: `You have liked accomodation`}
+        });
+    }
+    return res
+      .status(200)
+      .json({
+        status: 200,
+        data:{
+        message: `You have unliked an accomodation`}
+      });
+  } catch (error) {
+    return ApplicationError.internalServerError(error.message, res);
+  }
+};
+const disLike = async (req, res) => {
+  try {
+    const { accomodation_id: accomodationId } = req.params;
+    const { id: userId } = req.user;
+    const accomodationExist = await Accomodation.findOne({ where: { id: accomodationId } });
+    if (!accomodationExist) {
+      return ApplicationError.notFoundError(
+        'This accomodation does not exist',
+        res
+      );
+    }
+    const { created } = await UpdateOrCreate(Like, {accomodationId, userId}, false);  // UpdateOrCreate(model, {where and item})
+    if (created) {
+      return res
+        .status(201)
+        .json({
+          status: 201,
+          data:{
+          message: `You have disliked an accomodation`}
+        });
+    }
+    return res
+      .status(200)
+      .json({
+        status: 200,
+        data: {
+        message: `You have undisliked an accomodation`}
+      });
+  } catch (error) {
+    return ApplicationError.internalServerError(error.message, res);
+  }
+};
+const allLikes = async (req, res) => {
+  try {
+    const { accomodation_id: accomodationId } = req.params;
```

Comment:
validation. Use a middleware to validate accomodationId, one that can be reused for all this endpoints